File: frontend/src/services/api/supplierCommunicationService.js
import apiClient from './apiClient';

export const supplierCommunicationService = {
  getAllConversations: async (search = '', filter = '') => {
    const response = await apiClient.get('/supplier-communications/conversations', {
      params: { search, filter },
    });
    return response.data;
  },

  getSupplierConversation: async (supplierIdentifier) => {
    const response = await apiClient.get(`/supplier-communications/suppliers/${supplierIdentifier}/conversation`);
    return response.data;
  },

  getSupplierMessages: async (supplierIdentifier) => {
    const response = await apiClient.get(`/supplier-communications/suppliers/${supplierIdentifier}/messages`);
    return response.data;
  },

  sendMessageToSupplier: async (supplierIdentifier, content, messageType = 'SUPPLIER_MESSAGE', purchaseOrderId = null, attachmentFile = null) => {
    if (attachmentFile) {
      const formData = new FormData();
      formData.append('content', content);
      formData.append('messageType', messageType);
      if (purchaseOrderId) formData.append('purchaseOrderId', purchaseOrderId);
      formData.append('attachment', attachmentFile);

      const response = await apiClient.post(`/supplier-communications/suppliers/${supplierIdentifier}/messages`, formData, {
        headers: { 'Content-Type': 'multipart/form-data' },
      });
      return response.data;
    } else {
      const response = await apiClient.post(`/supplier-communications/suppliers/${supplierIdentifier}/messages/text`, {
        content,
        messageType,
        purchaseOrderId,
      });
      return response.data;
    }
  },

  startConversation: async (supplierId, purchaseOrderId = null, initialMessage = '') => {
    const response = await apiClient.post('/supplier-communications/conversations', {
      supplierId,
      purchaseOrderId,
      initialMessage,
    });
    return response.data;
  },

  updateConversationStatus: async (id, status) => {
    const response = await apiClient.put(`/supplier-communications/conversations/${id}/status`, {
      status,
    });
    return response.data;
  },

  getMyConversation: async () => {
    const response = await apiClient.get('/supplier-communications/conversations/my');
    return response.data;
  },

  getConversationById: async (id) => {
    const response = await apiClient.get(`/supplier-communications/conversations/${id}`);
    return response.data;
  },

  getConversationByPO: async (poId) => {
    const response = await apiClient.get(`/supplier-communications/by-po/${poId}`);
    return response.data;
  },

  getMessages: async (conversationId) => {
    const response = await apiClient.get(`/supplier-communications/conversations/${conversationId}/messages`);
    return response.data;
  },

  sendMessage: async (conversationId, content, messageType = 'SUPPLIER_MESSAGE', purchaseOrderId = null, attachmentFile = null) => {
    if (attachmentFile) {
      const formData = new FormData();
      formData.append('content', content);
      formData.append('messageType', messageType);
      if (purchaseOrderId) formData.append('purchaseOrderId', purchaseOrderId);
      formData.append('attachment', attachmentFile);

      const response = await apiClient.post(`/supplier-communications/conversations/${conversationId}/messages`, formData, {
        headers: { 'Content-Type': 'multipart/form-data' },
      });
      return response.data;
    } else {
      const response = await apiClient.post(`/supplier-communications/conversations/${conversationId}/messages/text`, {
        content,
        messageType,
        purchaseOrderId,
      });
      return response.data;
    }
  },

  markAsRead: async (conversationId) => {
    const response = await apiClient.put(`/supplier-communications/conversations/${conversationId}/read`);
    return response.data;
  },

  updatePOStatusFromChat: async (poId, status, note = '') => {
    const response = await apiClient.post(`/supplier-communications/purchase-orders/${poId}/update-status`, {
      status,
      note,
    });
    return response.data;
  },

  getUnreadCount: async () => {
    const response = await apiClient.get('/supplier-communications/unread-count');
    return response.data;
  },
};

File: backend/src/main/java/com/medistock/supplier/service/SupplierCommunicationService.java
package com.medistock.supplier.service;

import com.medistock.supplier.dto.CreateSupplierMessageRequest;
import com.medistock.supplier.dto.SupplierConversationResponse;
import com.medistock.supplier.dto.SupplierMessageResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface SupplierCommunicationService {
    List<SupplierConversationResponse> getAllConversationsForAdmin(String search, String filter);
    SupplierConversationResponse getConversationForSupplierUser(String userEmail);
    SupplierConversationResponse getOrCreateConversation(Long supplierId);
    SupplierConversationResponse getConversationBySupplierIdentifier(String supplierIdentifier, boolean createIfAbsent);
    SupplierConversationResponse startNewConversation(Long supplierId, Long purchaseOrderId, String initialMessage, String userEmail);
    SupplierConversationResponse updateConversationStatus(Long conversationId, String status);
    SupplierConversationResponse getConversationByPOId(Long purchaseOrderId);
    List<SupplierMessageResponse> getMessages(Long conversationId, String userEmail);
    List<SupplierMessageResponse> getMessagesBySupplierIdentifier(String supplierIdentifier, String userEmail);
    SupplierMessageResponse sendMessage(Long conversationId, CreateSupplierMessageRequest request, MultipartFile attachment, String userEmail);
    SupplierMessageResponse sendMessageBySupplierIdentifier(String supplierIdentifier, CreateSupplierMessageRequest request, MultipartFile attachment, String userEmail);
    void markAsRead(Long conversationId, String userEmail);
    SupplierMessageResponse updatePOStatusFromChat(Long purchaseOrderId, String newStatus, String note, String userEmail);
    void logPurchaseOrderStatusChangeEvent(com.medistock.purchase.entity.PurchaseOrder po, String oldStatus, String newStatus, String note, String userEmail);
    long getUnreadCount(String userEmail);
}
